File: easy_32_factorial_using_loop/solution.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    int computeFactorial(int n) {
        int result = 1;
        for (int i = 2; i <= n; ++i) {
            result *= i;
        }
        return result;
    }
};

int main() {
    int n;
    cin >> n;
    Solution solution;
    int result = solution.computeFactorial(n);
    cout << result << endl;
    return 0;
}

File: easy_32_factorial_using_loop/solution.py
from typing import TextIO
import sys

class Solution:
    def computeFactorial(self, n: int) -> int:
        """
        Returns the factorial of n using a loop.
        """
        result = 1
        for i in range(2, n + 1):
            result *= i
        return result

def solve(infile: TextIO = sys.stdin, outfile: TextIO = sys.stdout) -> None:
    n = int(infile.read().strip())
    solution = Solution()
    result = solution.computeFactorial(n)
    outfile.write(f"{result}\n")

if __name__ == "__main__":
    solve()
